### PR TITLE
fix: prefs empty array case

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

- fallback check to account for an empty array being returned was not accessiing possible null values correctly

## Test Plan

- manual

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

✅

### CODE REVIEW

File: src/lib/helpers/notifications.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ const notificationPrefs = (): Record<string, NotificationPrefItem> => {
     }
 
     // for some reason, the prefs become array as default or on all clear. let's reset.
-    return Array.isArray(prefs.notificationPrefs) ? {} : prefs.notificationPrefs || {};
+    return Array.isArray(prefs?.notificationPrefs) ? {} : (prefs?.notificationPrefs ?? {});
```
Comment (stnguyen90) [inline src/lib/helpers/notifications.ts:27]: There's already a null check above, though 🧐

Comment (stnguyen90) [reply src/lib/helpers/notifications.ts:27]: Ahhh it's a null check above...maybe that check should also check for undefined?

File: src/lib/helpers/notifications.ts [hunk 2]
```diff
@@ -19,12 +19,12 @@ const userPreferences = () => get(user)?.prefs;
 const notificationPrefs = (): Record<string, NotificationPrefItem> => {
     const prefs = userPreferences();
 
-    if (prefs === null) {
+    // due to php backend, empty object can be returnd as an empty array
+    if (prefs?.notificationPrefs === null || Array.isArray(prefs?.notificationPrefs)) {
```
Comment (stnguyen90) [inline src/lib/helpers/notifications.ts:23]: what if `prefs` is `undefined`?

Comment (TorstenDittmann) [reply src/lib/helpers/notifications.ts]: also fixed 👍🏻 improved the condition handling

**Review by stnguyen90 — APPROVED** [review]

**Pull request merged**